### ISSUE TITLE
Update to bedroom Substitution.

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#790]
_Imported from trac ticket 790.
Created by **rneville**
Opened in trac: Mar 7, 2011, 10:26:50 AM
Last modified in trac: Mar 18, 2011, 2:39:09 PM_
***
Kent or Rob,

Good morning, the following is an update to the bedroom substitution:

Mirrors are in the following minor and categories and hence a substitution
can occur as long as it is the same minor, vendor and family:

(MNR_CD = ‘400’ and CAT_CD IN (‘408’, ‘410’, ‘412’, ‘413’, ‘477’, ‘478’)) OR

(MNR_CD = 861’ and CAT_CD IN (‘864’, ‘865’, ‘874H’, ‘875C’))

Dressers are in the following minor and categories and hence a substitution
can occur as long as it is the same minor, vendor and family:

(MNR_CD = ‘400’ and CAT_CD IN (‘400’, ‘402’, ‘403’, ‘404’, ‘406’)) OR

(MNR_CD = 861’ and CAT_CD IN (‘862’, ‘863’))

Chest are in the following minor and categories and hence a substitution can
occur as long as it is the same minor, vendor and family:

(MNR_CD = ‘400’ and CAT_CD IN (‘410’, ‘411’, ‘414’, ‘415’, ‘416’, ‘417’,
‘419’, ‘421’, ‘423’, ‘424’, ‘425’, ‘480’, ‘494’)) OR

(MNR_CD = 861’ and CAT_CD IN (‘866’, ‘867’))

Headboards are in the following minor and categories and hence a
substitution can occur as long as it is the same minor, vendor and family:

(MNR_CD = ‘400’ and CAT_CD IN (‘427’, ‘438’, ‘441’, ‘443’, ‘444’, ‘445’,
‘446’, ‘447’, ‘449’, ‘451’, ‘452’, ‘473’, ‘489’, ‘490A’, ‘492A’, ‘495B’)) OR

(MNR_CD = 861’ and CAT_CD IN (‘873’, ‘873A’, ‘873D’, ‘874’, ‘874A’, ‘874C’,
‘874F’, ‘875’, ‘875A’))

Footboards are in the following minor and categories and hence a
substitution can occur as long as it is the same minor, vendor and family:

(MNR_CD = ‘400’ and CAT_CD IN (‘405’, ‘427’, ‘428’, ‘448’, ‘450’, ‘476’,
‘489’, ‘490A’, ‘495B’)) OR

(MNR_CD = 861’ and CAT_CD IN (‘873B’, ‘873E’, ‘874B’, ‘874D’))

Nightstands are in the following minor and categories and hence a
substitution can occur as long as it is the same minor, vendor and family:

(MNR_CD = ‘400’ and CAT_CD IN (‘466’, ‘467’)) OR

(MNR_CD = 861’ and CAT_CD IN (‘879’))

Thanks,

*Wayne Liburd*

*Director of MIS*

*Kanes Furniture Corporation*

5700 70th Ave North

Pinellas Park FL 33781

Tel: +1 (727) 545-9555 ext. 350

Fax: +1 (727) 541-6960

***
* kbower attached [moz-screenshot-66.png](https://trac.retailarchitects.com/trac/attachment/ticket/790/moz-screenshot-66.png) on Mar 8, 2011, 9:20:04 AM
* kbower attached [moz-screenshot-67.png](https://trac.retailarchitects.com/trac/attachment/ticket/790/moz-screenshot-67.png) on Mar 8, 2011, 9:20:05 AM
* kbower attached [moz-screenshot-68.png](https://trac.retailarchitects.com/trac/attachment/ticket/790/moz-screenshot-68.png) on Mar 8, 2011, 9:20:05 AM
* kbower attached [moz-screenshot-69.png](https://trac.retailarchitects.com/trac/attachment/ticket/790/moz-screenshot-69.png) on Mar 8, 2011, 9:20:05 AM
* kbower attached [moz-screenshot-70.png](https://trac.retailarchitects.com/trac/attachment/ticket/790/moz-screenshot-70.png) on Mar 8, 2011, 9:20:05 AM
* kbower attached [moz-screenshot-71.png](https://trac.retailarchitects.com/trac/attachment/ticket/790/moz-screenshot-71.png) on Mar 8, 2011, 9:20:06 AM

**Comment by kbower** [comment]
_**kbower** trac comment on Mar 8, 2011, 8:57:40 AM:_

Kent,

Currently the rule should be same MNR_CD, VE_CD, FAMILY_CD, CAT_CD, however because mirrors, dressers, chests, Headboards, footboards and nightstands can be in several categories and I do not believe they were defined.

Thanks,

*Wayne Liburd*


*From:*Kent Bower [mailto:kb@retailarchitects.com]
*Sent:* Tuesday, March 08, 2011 8:30 AM
*To:* Wayne Liburd
*Subject:* Re: Update to bedroom Substitution.

Wayne, for clarification: have the rules themselves changed at all or is this just update of the list of mnr_cd and cat_cd?

Kent

**Comment by kbower** [comment]
_**kbower** trac comment on Mar 8, 2011, 9:20:04 AM:_

These are not exactly the same definitions as those currently in the Rules.  (Although they are not being used in the rules, they are defined).

Please compare them and tell us only what needs to change.

Here is how you find what we have defined in Kane's rules:
Admin:


System Configuration:


Rules:


Select the "System" Rule Class:


Find the "app_startup" Rule:


Scroll over half way down the Rule and you'll find the definitions that look like this:



Please let us know if these definitions are already correct or if the ones you sent should /replace/ them or /add /to them, etc.
In other words, let us know */what needs to change /*from the current definitions.

Thanks,
Kent

[[Image(moz-screenshot-66.png)]]

[[Image(moz-screenshot-67.png)]]

[[Image(moz-screenshot-68.png)]]

[[Image(moz-screenshot-69.png)]]

[[Image(moz-screenshot-70.png)]]

[[Image(moz-screenshot-71.png)]]

**Comment by kbower** [comment]
_**kbower** trac comment on Mar 8, 2011, 12:32:25 PM:_

Ok, we will then add Chests, Headboards, Footboards and Nightstands.

We have not implemented any of the new requirements that you sent yesterday yet, so that is why your substitution failed....

(I wanted to make sure we had the definitions correct.)

-------- Original Message --------
Subject:         RE: #790: Update to bedroom Substitution.
Date:         Tue, 8 Mar 2011 12:24:20 -0500
From:         Wayne Liburd <wlibur@kanesfurniture.com>
To:         Kent Bower <kb@retailarchitects.com>
CC:         Rob Neville <rn@retailarchitects.com>


Kent,

Mirrors and Dressers are already defined so we need to define Chests, Headboards, Footboards and Nightstands. However, when trying to substitute a mirror 413199957 for 412199956 or a dresser 400199952 for 406199958 in package PALMCT10 in a package I get an error and based on the substitution rules this should work. Are all of the substitution rules you have received not active?

Thanks,

*Wayne Liburd*

**Comment by kbower** [comment]
_**rneville** trac comment on Mar 9, 2011, 4:11:45 PM:_

Wayne,

I'm trying to reconcile several conversations surrounding this and I'm
worried that the substitution rules in general may be wrong depending on
your answer.

You said:


And in another conversation, you said:


*QUESTION:*

Package Component
------------------------
minor = 400
vendor = 'VENDOR1'
family = 'FAMILY1'
cat_cd = 408

Substituted with
------------------------
minor = 861
vendor = 'VENDOR1'
family = 'FAMILY1'
cat_cd = 875C


*is this a legal substitution?*

**Comment by kbower** [comment]
_**wayne** trac comment on Mar 9, 2011, 4:26:55 PM:_

Rob,

 

This is not a legal substitution as it has to be the same MNR_CD.

 

Thanks,

 

Wayne Liburd

Director of MIS

Kanes Furniture Corporation

5700 70th Ave North

Pinellas Park FL 33781

Tel: +1 (727) 545-9555 ext. 350

Fax: +1 (727) 541-6960

From: Rob Neville [mailto:rn@retailarchitects.com] 
Sent: Wednesday, March 09, 2011 4:10 PM
To: Wayne Liburd; Kent Bower
Cc: issues
Subject: #790: Update to bedroom rules...not the usual yadda yadda
please read carefully

 

Wayne,

 

I'm trying to reconcile several conversations surrounding this and I'm
worried that the substitution rules in general may be wrong depending on
your answer.

 

You said:

        "Mirrors are in the following minor and categories and hence a
substitution
        can occur as long as it is the same minor, vendor and family:
        (MNR_CD = '400' and CAT_CD IN ('408', '410', '412', '413',
'477', '478')) OR
        (MNR_CD = 861' and CAT_CD IN ('864', '865', '874H', '875C'))"

And in another conversation, you said:

        "Currently the rule should be same MNR_CD, VE_CD, FAMILY_CD,
CAT_CD, however because mirrors, dressers, chests, Headboards,
footboards and nightstands can be in several categories and I do not
believe they were defined."

QUESTION:

Package Component
------------------------
minor = 400
vendor = 'VENDOR1'
family = 'FAMILY1'
cat_cd = 408

Substituted with
------------------------
minor = 861
vendor = 'VENDOR1'
family = 'FAMILY1'
cat_cd = 875C

 

is this a legal substitution?

**Comment by kbower** [comment]
_**rneville** trac comment on Mar 9, 2011, 4:29:58 PM:_

None of the other substitution rules have such a restriction when given a
rule like that.

Is this one different or should ALL of them work like that?

**Comment by kbower** [comment]
_**rneville** trac comment on Mar 9, 2011, 4:39:05 PM:_

Actually, that's easy enough to do, my worry if for all the other rules. Are
you saying that in general it's probably OK?

**Comment by kbower** [comment]
_**rneville** trac comment on Mar 9, 2011, 5:37:38 PM:_

Begin forwarded message:

**Comment by kbower** [comment]
_**rneville** trac comment on Mar 10, 2011, 8:56:14 AM:_

(Changeset [[1296]](../commit/849d69edced9b865dbb30df2468940082697625d))
fixes #790 (Update to bedroom Substitution)

**Comment by kbower** [comment]
_**rneville** trac status change on Mar 10, 2011, 8:56:14 AM:_ **closed**
